core/src/main/java/org/mule/runtime/core/internal/execution/FlowProcessMediator.java

```diff
@@ -217,6 +219,10 @@ public void process(FlowProcessTemplate template,
       final CoreEvent event = createEvent(template, messageSource,
                                           responseCompletion, flowConstruct);
 
+      template.getSourceMessage().getNotification().ifPresent(n -> {
```

Comment:
The SourceResultAdapter class says that the notification is an Object, if the notification is not a PollingSourceItemNotification, it will fail. Also, if in the future we need to do the same with another kind of annotation, the getNotification methods falls short.